src/rgw/rgw_lua_utils.cc

```diff
@@ -85,10 +84,93 @@ void open_standard_libs(lua_State* L) {
   unsetglobal(L, "dofile");
   unsetglobal(L, "debug");
   // remove os.exit()
-  lua_getglobal(L, "os");
-  lua_pushstring(L, "exit");
-  lua_pushnil(L);
-  lua_settable(L, -3);
+  if (lua_getglobal(L, "os") == LUA_TTABLE) {
+    lua_pushstring(L, "exit");
+    lua_pushnil(L);
+    lua_settable(L, -3);
+  }
```

Comment:
keep os.execute?

Reply:
you are right. probably remove that as well, this would cause fork+exec that would probably block all RGW threads.
btw, in the future I plan on adding a "sandboxed lua" conf parameter that would prevent any external interactions.

Reply:
lua "hardening" requirements are captured here: https://tracker.ceph.com/issues/62120